elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -62,8 +62,9 @@ public ElasticsearchSchemaFactory() {
       final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
           new TypeReference<Map<String, Integer>>() { });
 
+      final String scheme = (String) map.get("scheme");
```

Comment:
Why scheme has to be hardcoded in configuration ? 
Can it be auto-detected programmatically ?

Reply:
Scheme can't be supplied in coordinates like:
 "coordinates": "{'https://vpc.xxx': 443}" otherwise it fails in later steps of ElasticsearchTransport. So the scheme has to be separately supplied.

Reply:
Now when I think, it seems better to introduce another config variable like `hosts` or `nodes` which will be a JSON array of URIs:
```
  hosts: ["http://127.0.0.1:443", "http://myNode:80"]
```

Reply:
Introduced new variable *hosts* in parallel to coordinates to maintain backward compatibility.

Either hosts or coordinates is required to connect to ES. If hosts is provided it should be a well-formed URL having protocol, host and port information. Hosts can take any number of hosts.
With coordinates it works as before.